### DISCUSSION TITLE
Publish to https://open-vsx.org/

### DISCUSSION THREAD

**Discussion by kaashyapan** [ionide/ionide-vscode-fsharp#1406] [Ideas]
**Is your feature request related to a problem? Please describe.**
vscode is great and not a walled garden (but maybe a fenced garden ?)
I'd like to stay away from Microsoft as much as is possible considering I use FSharp.
Can we please also publish this extension to an open repo ?

**Describe the solution you'd like**
Publish this extension to https://open-vsx.org/ which is an open repository for vscode compatible IDEs like vscodium.

[a comment by an automated account is not shown]

**Comment by baronfel** [comment]
I certainly have no problems with publishing to other repositories. Do you have any pointers to documentation about how to push to that repository? Right now we publish as part of the CI pipeline, so if you could figure out how to push to it as part of a github action that would be ideal.

**Comment by kaashyapan** [comment]
This has a github action
[https://github.com/HaaLeo/publish-vscode-extension#readme](https://github.com/HaaLeo/publish-vscode-extension#readme)

The 'How-to' to publish an extension is in 
https://github.com/eclipse/openvsx/wiki/Publishing-Extensions

Create an account -> setup namespace -> Get a token -> Setup the above github action

**Comment by Krzysztof-Cieslak** [comment]
Sounds good to me, I think GitPod integrates open-vsx marketplace too

**Comment by OmanF** [comment]
Another promising editor using `open-vsx` exclusively, due to legal issues, is [oni2](https://github.com/onivim/oni2) (previously `onivim`).

Would be great having Ionide there too for all VSCode-compatible editors.

Thanks.

**Comment by olivercoad** [comment]
+1 to publishing to open-vsx, but just wondering how does https://github.com/ionide/playground work in Gitpod currently?

**Comment by nirvinm** [comment]
I would love to install Ionide from open-vsix.org. But how to handle C# extension dependency given that it is only licensed to run under Microsoft's VSCode build and they are never going to make it available on open-vsix?

**Comment by muhammadsammy** [comment]
I created a free fork 
<https://github.com/muhammadsammy/free-omnisharp-vscode> of the 
official C# extension that can be used as the C# dependency for Ionide 
on Open VSX <https://open-vsx.org/extension/muhammad-sammy/csharp>. I 
tried it for debugging a simple F# console and it worked fine.

> I would love to install Ionide from open-vsix.org. But how do we 
> handle C# extension dependency on open-vsx given that C# extension is 
> only licensed to run under Microsoft's VSCode build.
>

**Reply by nojaf** [reply]
Great, so what needs to be done now? A setting in Ionide?

**Reply by muhammadsammy** [reply]
I installed the vsix file then changed the C# extension ID from `ms-dotnettools` to `muhammad-sammy`

 in:
`$HOME/.vscode-oss/extensions/ionide.ionide-*/.vsixmanifest`
and
`$HOME/.vscode-oss/extensions/ionide.ionide-*/package.json`

Maybe this could be accomplished with a GitHub Action that runs a script to change the C# dependency only for Ionide on OpenVSX, then releases it to OpenVSX.

**Reply by nojaf** [reply]
It looks indeed like changing https://github.com/ionide/ionide-vscode-fsharp/blob/ab871e5bcfec001bbc81ef5b331fbc0bb732b1b8/release/package.json#L1568 to `"muhammad-sammy.csharp"`.
I might be able to cook something up in the FAKE script.

At some point, one of the maintainers will need to do the registration thing at https://github.com/eclipse/openvsx/wiki/Publishing-Extensions#1-create-an-eclipse-account

**Reply by nojaf** [reply]
Instead of the entire csharp package, would using https://github.com/redhat-developer/netcoredbg-theia-plugin be an option?
It has the Samsung debugger as well and looks like a lightweight solution than the entire csharp extension.
@baronfel any ideas?

**Comment by kaashyapan** [comment]
Some updates from Onivim2 ...  

> The Ionide.Ionide-fsharp extension depends on `ms-dotnettools.csharp`, which isn't available on open-vsx


https://github.com/onivim/oni2/commit/7929ea58418b698de441801883d56fe2501021af
